### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by olawale-o**
## In this PR, I did the following:

[Loom Link](https://www.loom.com/share/0da622965ee5452fafee311aff139fc0)
- Splash screen

  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages

  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)

  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that     category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
  
- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page

  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page

  - The user fills out a form to create a new transaction with:
    - name (mandatory)
   - amount (mandatory)
  - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
   - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
 - Testing requirements
    - Create unit and integration tests for all the most important components of your RoR application.

## Tip:
- For logo url to use you can checkout this collections [link](https://seeklogo.com/search?q=food+and+drinks&page=3)

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      <% else %>
+        <% @groups.each do |group| %>
+          <%= render partial: 'group_card', locals: { group: group } %>
+        <% end %>
+      <% end %>
+    </ul>
+  </div>
+
+  <div class="cta__container">
+    <%= link_to "ADD A NEW CATEGORY", new_category_path, class: "add__btn" %>
```
Comment (arslanbisharat) [inline app/views/categories/index.html.erb:28]: - [ ] On adding category it throws an error and then I cannot move forward. 
Capture

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ] Make sure you are following correct Gitflow.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/views/categories/index.html.erb (hunk 1 above)
Comment (olawale-o) [reply app/views/categories/index.html.erb:28]: Hi @arslanbisharat 
Thank you for your time and effort on the review
I guess you tried the deployed link. I just added a new category it is working as it should. kindly find below a snapshot of the working one
Screen Shot 2021-11-22 at 21 06 06

**Review by od-c0d3r — APPROVED** [review]
Asslam Alaikum @olawale-o ,😊

Hi👋🏽, 
It has been long a journey for you..

> "Successful people do what unsuccessful people are not willing to do. Don't wish it were easier; wish you were better."
>-Jim Rohn

Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

#### All requirements has been satisfied and am happily saying that Your project is complete. :100: You can now merge it :sunglasses: :ok_hand:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by olawale-o** [comment]
Hi @od-c0d3r 
Thank you for your time and effort on the review.

I really appreciate the quote from "Jim Rohn"

**Pull request merged**